com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSlider.cs

```diff
@@ -302,28 +305,24 @@ public PiecewiseLightUnitSlider(LightUnitSliderUIDescriptor descriptor) : base(d
                 piece.domain = new Vector2(x0, x1);
                 piece.range  = new Vector2(y0, y1);
 
-                piece.transform = GetTransformation(x0, x1, y0, y1);
+                NewTransformation(x0, x1, y0, y1, out piece.directM, out piece.directB);
 
                 // Compute the inverse
-                CoreUtils.Swap(ref x0, ref y0);
-                CoreUtils.Swap(ref x1, ref y1);
-                piece.inverseTransform = GetTransformation(x0, x1, y0, y1);
+                NewTransformation(y0, y1, x0, x1, out piece.inverseM, out piece.inverseB);
 
-                var k = sortedRanges[i].value.GetHashCode();
-                m_PiecewiseFunctionMap.Add(k, piece);
+                m_PiecewiseFunctionMap.Add(sortedRanges[i].value, piece);
             }
         }
 
         protected override float GetPositionOnSlider(float value, Vector2 valueRange)
         {
-            var k = valueRange.GetHashCode();
-            if (!m_PiecewiseFunctionMap.TryGetValue(k, out var piecewise))
+            if (!m_PiecewiseFunctionMap.TryGetValue(valueRange, out var piecewise))
                 return -1f;
 
             return ValueToSlider(piecewise, value);
         }
 
-        void UpdatePiece(ref Piece piece, float x)
+        bool UpdatePiece(ref Piece piece, float x)
```

Comment:
Maybe indicate a small comment about the semantic of the returned boolean and that it performs a piece search based on `x` value.

It is not very obvious what kind of update is expected to be performed.